img/service.go

```diff
@@ -183,3 +197,46 @@ func (s *Service) detectFormat(in io.Reader) (Format, io.Reader, error) {
 
 	return format, io.MultiReader(buf, in), nil
 }
+
+func getEmbeddedThumbnail(in io.Reader) ([]byte, io.Reader, error) {
+	buf := &bytes.Buffer{}
+	r := io.TeeReader(in, buf)
+	wrappedReader := io.MultiReader(buf, in)
+
+	offset := 0
+	offsets := []int{12, 30}
```

Comment:
Can you please explain why are you checking those offsets only?

Reply:
Those are the two most common offsets where the Exif starts in JPEG files. To avoid more complex parsing (and library dependencies), it only checks those offsets.

Reply:
I checked that in the provided library the different approach is used https://github.com/dsoprea/go-exif/blob/d154f10435cc/v3/exif.go#L52

Reply:
Indeed. And that's the approach in the first (unpublished) attempt. That function tries to find an exif in brute-force fashion byte by byte scanning the whole file: https://github.com/dsoprea/go-exif/blob/d154f10435cc6ae4b64cd68eed04a1d005b73ec7/v3/exif.go#L150

It is more efficient by just checking the most common offsets for exif and there was a noticeable performance increase.